platform/ios/src/MGLMapViewDelegate.h

```diff
@@ -382,6 +382,27 @@ NS_ASSUME_NONNULL_BEGIN
 - (void)mapView:(MGLMapView *)mapView didDeselectAnnotation:(id <MGLAnnotation>)annotation;
 
 /**
+ Tells the delegate that one of its shape annotations was selected.
+ 
+ You can use this method to track changes in the selection state of annotations.
+ 
+ @param mapView The map view containing the annotation.
+ @param shapeAnnotation The shape annotation that was selected.
+ */
+- (void)mapView:(MGLMapView *)mapView didSelectShapeAnnotation:(MGLShape *)shapeAnnotation;
```

Comment:
Given that the existing methods are called `-mapView:didSelectAnnotation:` and `-mapView:didDeselectAnnotation:`, without referring to point annotations specifically, it’s interesting that this PR adds separate shape-specific methods.

On the one hand, I understand that suddenly passing shape annotations into a method that has long received only point annotations could be a surprise to some developers. On the other hand, we’ve never documented that prior limitation – it’s essentially a bug. There is precedent for opening up delegate methods to new data, namely when we started passing MGLUserLocation into `-mapView:viewForAnnotation:` in #5882.

Reply:
In https://github.com/mapbox/mapbox-gl-native/pull/9755#discussion_r136166060 I mention an alternative

Reply:
The reason behind naming this as `didSelectShapeAnnotation` is based on been clear to developers that this is not going to be called by selecting any *MGLPointAnnotation thus there is no callout view.